.gitignore

```diff
@@ -31,6 +31,9 @@ codegen/protocol-test-codegen-local/smithy-build.json
 codegen/protocol-test-codegen/smithy-build.json
 SmokeTests/
 
+# Preview embedded dependencies
+/smithy-swift/
+
```

Comment:
We'll be embedding `smithy-swift` in the SDK repo in preview build artifacts.  This `.gitignore` will make it so that if a preview customer wants to push their SDK up to Git, `smithy-swift` won't get pushed with it.